#!/usr/bin/env python
PACKAGE = "fake_publisher"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("r1_error", double_t, 0, "A double parameter",    0.0, -0.5,   0.5)
gen.add("r2_error", double_t, 0, "A double parameter",    0.0, -0.5,   0.5)
gen.add("r3_error", double_t, 0, "A double parameter",    0.0, -0.5,   0.5)
gen.add("r4_error", double_t, 0, "A double parameter",    0.0, -0.5,   0.5)
gen.add("r5_error", double_t, 0, "A double parameter",    0.0, -0.5,   0.5)
gen.add("r6_error", double_t, 0, "A double parameter",    0.0, -0.5,   0.5)
gen.add("r7_error", double_t, 0, "A double parameter",    0.0, -0.5,   0.5)
gen.add("r8_error", double_t, 0, "A double parameter",    0.0, -0.5,   0.5)

gen.add("a_error", double_t, 0, "A double parameter",    0.0, -0.5,   0.5)
gen.add("b_error", double_t, 0, "A double parameter",    0.0, -0.5,   0.5)
gen.add("c_error", double_t, 0, "A double parameter",    0.0, -0.5,   0.5)
gen.add("d_error", double_t, 0, "A double parameter",    0.0, -0.5,   0.5)
gen.add("e_error", double_t, 0, "A double parameter",    0.0, -0.5,   0.5)
gen.add("f_error", double_t, 0, "A double parameter",    0.0, -0.5,   0.5)
gen.add("g_error", double_t, 0, "A double parameter",    0.0, -0.5,   0.5)
gen.add("h_error", double_t, 0, "A double parameter",    0.0, -0.5,   0.5)

gen.add("RSS0", double_t, 0, "A double parameter",    0.0, 0.0,   90.0)
gen.add("RSS1", double_t, 0, "A double parameter",    0.0, 0.0,   90.0)
gen.add("RSS2", double_t, 0, "A double parameter",    0.0, 0.0,   90.0)
gen.add("RSS3", double_t, 0, "A double parameter",    0.0, 0.0,   90.0)
gen.add("RSS4", double_t, 0, "A double parameter",    0.0, 0.0,   90.0)
gen.add("RSS5", double_t, 0, "A double parameter",    0.0, 0.0,   90.0)
gen.add("RSS6", double_t, 0, "A double parameter",    0.0, 0.0,   90.0)
gen.add("RSS7", double_t, 0, "A double parameter",    0.0, 0.0,   90.0)

gen.add("min_distancia", double_t, 0, "A double parameter",    0.0, -0.5,   0.0)
gen.add("max_distancia", double_t, 0, "A double parameter",    0.0, 0.0,   0.5)

gen.add("min_pos", double_t, 0, "A double parameter",    0.0, -0.5,   0.0)
gen.add("max_pos", double_t, 0, "A double parameter",    0.0, 0.0,   0.5)

gen.add("media_range", double_t, 0, "A double parameter",    0.0, 0.0,   0.5)
gen.add("desviacion_tipica_range", double_t, 0, "A double parameter",    0.0, 0.01,   0.2)

gen.add("media_posicion", double_t, 0, "A double parameter",    0.0, 0.0,   0.5)
gen.add("desviacion_tipica_posicion", double_t, 0, "A double parameter",    0.0, 0.01,   0.2)

gen.add("random",   bool_t,   0, "A Boolean parameter",  True)
gen.add("uniforme",   bool_t,   0, "A Boolean parameter",  True)

exit(gen.generate(PACKAGE, "fake_publisher", "Error"))
